### PR TITLE
Task-3, 4 and 5 for javascript are added

### PR DESCRIPTION
**Pull request by krushna-dongre**

### CODE REVIEW

File: Assignments/Task-5/array_func.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Array Helpers
+// http://www.codewars.com/kata/array-helpers
+
+Array.prototype.square = function() {
```
Comment (Ramkrish1214) [inline Assignments/Task-5/array_func.js]: before adding any method to prototype, please verify if that method is already exists on prototype or not

File: Assignments/Task-5/nested_obj.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Extract Nested Object Reference
+// http://www.codewars.com/kata/extract-nested-object-reference
+
+Object.prototype.hash = function(string) {
```
Comment (Ramkrish1214) [inline Assignments/Task-5/nested_obj.js]: Please follow the same as per above comment

File: Assignments/Task-3/PrefillAnArray.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//https://www.codewars.com/kata/54129112fb7c188740000162/train/javascript
+
+
+function prefill(n, v) {
+ 
+  if(isNaN(Number.parseInt(n)) ||(n % 1 !== 0) || n < 0)
```
Comment (Ramkrish1214) [inline Assignments/Task-3/PrefillAnArray.txt]: wrap all if and else blocks with braces - { }

File: Assignments/Task-4/cache_function.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// Function Cache
+// http://www.codewars.com/kata/a-chain-adding-function
+
+function cache(func) {
+    var memory = {};
+    return function() {
+        var key = JSON.stringify(arguments);
+        if (key in memory) return memory[key];
+        return memory[key] = func.apply(this, arguments);
```
Comment (Ramkrish1214) [inline Assignments/Task-4/cache_function.js]: Please wrap if block code in braces { }
and assign memory[key] and then return  memory[key]
And never assign any value in return statement directly.

**Review by Ramkrish1214 — COMMENTED** [review]
once you fixed all the below changes please reply to the review comments by sating its comepleted

File: Assignments/Task-5/array_func.js (hunk 1 above)
Comment (krushna-dongre) [reply Assignments/Task-5/array_func.js]: It's completed

File: Assignments/Task-5/nested_obj.js (hunk 2 above)
Comment (krushna-dongre) [reply Assignments/Task-5/nested_obj.js]: Changes are done

File: Assignments/Task-3/PrefillAnArray.txt (hunk 3 above)
Comment (krushna-dongre) [reply Assignments/Task-3/PrefillAnArray.txt]: Done

File: Assignments/Task-4/cache_function.js (hunk 4 above)
Comment (krushna-dongre) [reply Assignments/Task-4/cache_function.js]: Completed

**Pull request merged**